src/diapysef/scripts/high_precision_irt.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/env python
+
+# WRITE A COMMANDLINE TOOL FOR HIGH-PRECISION IRT LIBRARY GENERATION
+# FOR THE OPENSWATH tr_irt_nonlinear option
+
+import numpy as np
+import re
+import pandas as pd
+import os
+import sys
+
+if len(sys.argv) < 4:
+    print("Usage: high_precision_irt.py mqout_dir output_file peptide_number_cutoff")
+    sys.exit()
+
+mqout_dir = sys.argv[1]
+# mqout_dir can be 1 MQ output from multiple runs or a directory that holds
+# multiple MQ out files 
+
+outfile = sys.argv[2]
+# One single output high_precision_iRT.txt
+
+cutoff = int(sys.argv[3] )
+# Peptide number cutoff
+
+filenames = []
+for root, dirs, files in os.walk(mqout_dir):
+     for name in files:
+         #print(os.path.join(root, name))
+         filenames.append(os.path.join(root, name))
+
+r = re.compile(".*msms.txt")
+
+msms_files = list(filter(r.findall, filenames))
+
+msms = []
+sequence = []
+for i in msms_files:
+    stat = pd.read_csv(i, sep = '\t')
+    msms.append(pd.read_csv(i, sep = '\t'))
+    sequence.append(stat['Modified sequence'].to_list())
+
+# get the intersection peptides
+
+intersection_peptides = list(set.intersection(*map(set, sequence)))
+print(len(intersection_peptides))
+
+
+pep = []
+for i in msms: pep.append(i['PEP'].mean())
+
+msms_irt = msms[np.argmin(pep)]
+
+msms_irt = msms_irt.loc[msms_irt['Modified sequence'].isin(intersection_peptides)]
+
+msms_irt = msms_irt.sort_values("PEP", ascending = True)
+msms_irt = msms_irt.drop_duplicates(subset = "Modified sequence")
+msms_irt = msms_irt.rename(columns = {'Modified sequence' : "ModifiedPeptideSequence"})
+
+ev_file = os.path.join(os.path.dirname(msms_files[np.argmin(pep)]), 'evidence.txt')
+ev = pd.read_csv(ev_file, sep = '\t')
+ev = ev.loc[:, ["id", "Calibrated retention time", 'Calibrated 1/K0', 'Ion mobility index']]
```

Comment:
@klannieha this will not work for regular MQ output. can you just check if the column is there and then extract it or not?